BrainPortal/lib/cbrain_file_revision.rb

```diff
@@ -99,14 +96,32 @@ def self.[](fullpath)
     self.new(fullpath)
   end
 
-  # Returns the revision info in a fake 'svn' format:
-  #   "$Id: en_cbrain_local_data_provider.rb 12ab34 2010-03-25 17:26:05Z prioux $"
-  # Note that this triggers a self_update() call, if necessary, the first time.
   def to_s
-    return @fake_svn_id_string if ! @fake_svn_id_string.nil? # Cached; no need to fetch git info every time.
-    self_update()
-    @fake_svn_id_string = "$Id: #{@basename} #{@short_commit} #{@date} #{@time} #{@author} $"
-    @fake_svn_id_string
+    self.pretty({:commit => true, :author => true, :date => true})
+  end
+
+  def pretty(rev_fields = {})
```

Comment:
Method undocumented

Also: you decode some possible subsets of keys in the arguments?!? What is if subset is not in the if/else cases? (e.g. :file and :time only) ?